acp/phpbb_ideas_module.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ *
+ * Ideas extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\ideas\acp;
+
+class phpbb_ideas_module
+{
+	/** @var \phpbb\config\config */
+	protected $config;
+
+	/** @var \phpbb\db\driver\driver_interface */
+	protected $db;
+
+	/** @var \phpbb\log\log */
+	protected $log;
+
+	/** @var \phpbb\request\request */
+	protected $request;
+
+	/** @var \phpbb\template\template */
+	protected $template;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var string */
+	public $u_action;
+
+	public function main($id, $mode)
+	{
+		global $config, $db, $phpbb_log, $request, $template, $user;
+
+		$this->config = $config;
+		$this->db = $db;
+		$this->log = $phpbb_log;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+
+		// Add the board rules ACP lang file
+		$this->user->add_lang_ext('phpbb/ideas', 'phpbb_ideas_acp');
+
+		// Load a template from adm/style for our ACP page
+		$this->tpl_name = 'acp_phpbb_ideas';
+
+		// Set the page title for our ACP page
+		$this->page_title = 'ACP_PHPBB_IDEAS_SETTINGS';
```

Comment:
Put all of the above set-up in a constructor like: https://github.com/VSEphpbb/similartopics/blob/master/acp/similar_topics_module.php#L63-L91